src/nvme/ioctl.c

```diff
@@ -42,14 +44,116 @@ static int nvme_verify_chr(int fd)
 	return 0;
 }
 
+static bool nvme_ioctl_cmd_admin(unsigned long req)
+{
+	if (req == NVME_IOCTL_ADMIN_CMD || req == NVME_IOCTL_ADMIN64_CMD)
+		return true;
+
+	return false;
+}
+
+static bool nvme_ioctl_cmd_64(unsigned long req)
+{
+	if (req == NVME_IOCTL_ADMIN64_CMD || req == NVME_IOCTL_IO64_CMD)
+		return true;
+
+	return false;
+}
+
+static const char *nvme_cmd_to_string(unsigned long req, __u8 opcode)
+{
+	const char *cmd_name;
+
+	if (nvme_ioctl_cmd_admin(req))
+		cmd_name = nvme_admin_to_string(opcode);
+	else
+		cmd_name = nvme_nvm_to_string(opcode);
+
+	if (!cmd_name)
+		return "Unknown";
+
+	return cmd_name;
+}
+
+static void nvme_show_latency(unsigned long req, __u8 *opcode, struct timeval *start,
+			      struct timeval *end)
+{
+	struct timeval latency;
+
+	timersub(end, start, &latency);
+
+	if (opcode)
+		printf("%s%s Command opcode: %02x (%s) ",
+		       nvme_ioctl_cmd_admin(req) ? "Admin" : "IO",
+		       nvme_ioctl_cmd_64(req) ? "64" : "", *opcode,
+		       nvme_cmd_to_string(req, *opcode));
+	else
+		printf("ioctl: %lx (%s) ", req, nvme_ioctl_to_string(req));
+
+	printf("latency: %lu us\n", latency.tv_sec * 1000000 + latency.tv_usec);
+}
+
+static bool nvme_ioctl_cmd(unsigned long req)
+{
+	bool ioctl_cmd = false;
+
+	switch (req) {
```

Comment:
Do you really need to use `fallthrough`? Why not just:
```
	switch (req) {
	case NVME_IOCTL_ADMIN_CMD:
	case NVME_IOCTL_IO_CMD:
	case NVME_IOCTL_ADMIN64_CMD:
	case NVME_IOCTL_IO64_CMD:
		ioctl_cmd = true;
		break;
```

Reply:
The checkpatch.pl tool checks `fallthrough` and the tool is used by the PR check I think. Also usually used it in Kernel code I think also.
  <https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/scripts/checkpatch.pl#n7302>